internal/plugins/nfs-shares.go

```diff
@@ -276,6 +284,8 @@ func prometheusGetSingleValue(api prom_v1.API, queryStr string) (float64, error)
 
 	switch resultVector.Len() {
 	case 0:
+		//Note: this error message is used by assetManagerNFS.GetAssetStatus()
+		//to check if the share still exists in the backend.
 		return 0, fmt.Errorf("Prometheus query returned empty result: %s", queryStr)
```

Comment:
Since the caller matches on this error message to invoke special behavior, this should be its own error type. Type assertions are more robust than string matches.